src/Raven.Client/Documents/Session/Operations/StreamOperation.cs

```diff
@@ -403,20 +412,20 @@ public async ValueTask<bool> MoveNextAsync()
                     return false;
                 }
 
-                using (var builder = new BlittableJsonDocumentBuilder(_session.Context, BlittableJsonDocumentBuilder.UsageMode.ToDisk, "readArray/singleResult", _parser, _state))
+                _builder = new BlittableJsonDocumentBuilder(_builderContext, BlittableJsonDocumentBuilder.UsageMode.ToDisk, "readArray/singleResult", _parser, _state);
```

Comment:
it is creating builder every time in contrast to `MoveNext`

Reply:
That is the same behavior as before, no? But where is it closed?
Did you mean: 

```suggestion
                _builder ??= new BlittableJsonDocumentBuilder(_builderContext, BlittableJsonDocumentBuilder.UsageMode.ToDisk, "readArray/singleResult", _parser, _state);
```